packages/mui-material/src/CardHeader/CardHeader.js

```diff
@@ -98,59 +99,92 @@ const CardHeader = React.forwardRef(function CardHeader(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
+  const externalForwardedProps = {
+    slots,
+    slotProps: {
+      ...slotProps,
+      title: slotProps.title ?? titleTypographyProps,
+      subheader: slotProps.subheader ?? subheaderTypographyProps,
+    },
+  };
+
   let title = titleProp;
+  const [TitleSlot, titleSlotProps] = useSlot('title', {
+    className: classes.title,
+    elementType: Typography,
+    externalForwardedProps,
+    ownerState,
+  });
   if (title != null && title.type !== Typography && !disableTypography) {
     title = (
-      <Typography
-        variant={avatar ? 'body2' : 'h5'}
-        className={classes.title}
-        component="span"
-        {...titleTypographyProps}
-      >
+      <TitleSlot variant={avatar ? 'body2' : 'h5'} component="span" {...titleSlotProps}>
```

Comment:
Should we move `variant` and `component` to the `useSlot` hook as well?

Reply:
done.